File: alura/php/solid/ContaComum.php
<?php
  class ContaComum {

    private $manipulador;

    public function __construct() {
      $this->manipulador = new ManipuladorDeSaldo();
    }

    public function saca($valor) {
      // regra de negócio
      $this->manipulador->saca($valor);
    }

    public function deposita($valor) {
      $this->manipulador->deposita($valor);
    }

    public function getSaldo() {
      return $this->manipulador->getSaldo();
    }

    public function rende() {
      $this->manipulador->rende(1.1);
    }
  }
?>

File: alura/c-sharp/solid/ContaComum.cs
namespace Solid
{
  public class ContaComum
  {
    private ManipuladorDeSaldo manipulador

    public ContaComum()
    {
    this.manipulador = new ManipuladorDeSaldo();
    }

    public void Saca(double valor)  {
      manipulador.Saca(valor);
    }

    public void Deposita(double valor)  {
      manipulador.deposita(valor);
    }

    public void SomaInvestimento()  {
      this.manipulador.SomaInvestimento(1.1);
    }
  }
}
